packages/docusaurus-preset-classic/src/index.ts

```diff
@@ -74,13 +76,16 @@ export default function preset(
       makePluginConfig('@docusaurus/plugin-google-analytics', googleAnalytics),
     );
   }
-  if (debug || (debug === undefined && !isProd)) {
+  if (isDebugEnabled) {
     plugins.push(require.resolve('@docusaurus/plugin-debug'));
   }
   if (gtag) {
     plugins.push(makePluginConfig('@docusaurus/plugin-google-gtag', gtag));
   }
   if (isProd && sitemap !== false) {
+    if (isDebugEnabled) {
+      (sitemap.ignorePatterns ||= []).push(`/${debugPluginRouteBasePath}/**`);
```

Comment:
Oh, and also, our `ignorePatterns` requires you to prepend the base URL yourself (because I don't know if naively prepending `/baseUrl/` to all glob patterns is safe), so we need to do that here as well.